climada/engine/test/test_impact.py

```diff
@@ -868,6 +887,37 @@ def test_select_imp_map_fail(self):
         with self.assertRaises(ValueError):
             imp.select(event_ids=[0], event_names=[1, 'two'], dates=(0, 2))
 
+    def test_select_reset_frequency(self):
+        """Test that reset_frequency option works correctly"""
+
+        imp = dummy_impact_yearly() # 6 events, 1 per year
+
+        # select first 4 events
+        n_yr = 4
+        sel_imp = imp.select(dates=(imp.date[0],imp.date[n_yr-1]), reset_frequency=True)
+
+        # check unchanged attributes
+        self.assertTrue(u_coord.equal_crs(sel_imp.crs, imp.crs))
+        self.assertEqual(sel_imp.unit, imp.unit)
+        self.assertEqual(sel_imp.frequency_unit, imp.frequency_unit)
+        np.testing.assert_array_equal(sel_imp.coord_exp, imp.coord_exp)
+
+        # check sub-selected attributes
+        np.testing.assert_array_equal(sel_imp.event_id, imp.event_id[0:n_yr])
+        self.assertEqual(sel_imp.event_name, imp.event_name[0:n_yr])
+        np.testing.assert_array_equal(sel_imp.date, imp.date[0:n_yr])
+        np.testing.assert_array_equal(sel_imp.at_event, imp.at_event[0:n_yr])
+        np.testing.assert_array_equal(sel_imp.imp_mat.todense(),
+                                      imp.imp_mat[0:n_yr,:].todense())
```

Comment:
Not sure if this should be part of the particular test 🤔 It should already be part of the other tests, and we do not need to double that.

Reply:
Good point. As these are not directly influenced by the reset_frequency option I removed these lines.